framework/src/service/ServiceListeners.cpp

```diff
@@ -446,7 +446,24 @@ void ServiceListeners::GetMatchingServiceListeners(const ServiceEvent& evt,
       const LDAPExpr& ldapExpr = sse.GetLDAPExpr();
       if (ldapExpr.IsNull() || ldapExpr.Evaluate(props, false)) {
         set.insert(sse);
-      }
+      } else if (evt.GetType() == ServiceEvent::SERVICE_REGISTERED) {
+        // get the props key and value for logging
+        std::string propsKeyValues{};
+
+        auto key = props->Keys_unlocked();
+        for (auto k : key) {
+          auto val = props->Value_unlocked(k).ToString();
+          propsKeyValues += k + " : " + val + ", ";
+        }
+
+        const std::string msg =
+          "Service listener matching LDAP filter could "
+          "not be found using LDAP filter: " +
+          sse.GetFilter() + ". Please verify the LDAP filter value in service properties: " + "{" + propsKeyValues + "}";
+
+          coreCtx->logger->Log(
+            cppmicroservices::logservice::SeverityLevel::LOG_DEBUG, msg);
```

Comment:
There are many cases where not finding the matching service listener is the normal case. Just wondering if this is the right place to be logging this. The normal cases will be cluttering up the log. Would it make more sense to have the caller of GetMatchingServiceListeners do the logging? They are the ones that know when not finding a match is normal or not.

Reply:
I think it's normal for someone to use an LDAP filter to search for services with a specific service property. It's not an error if they don't succeed in finding a service. It just means the service hasn't been registered yet. Wouldn't this logging code log those cases as well?

Reply:
yes, it will log in those cases. the problem is, we don't know what the intent is. the goal is to log something for users to see in the log files when it's the case that they didn't find the service they expected due to a typo in a LDAP filter and/or service property. the downside is that there are more log entries.